hikari/channels.py

```diff
@@ -101,12 +102,89 @@ def __str__(self) -> str:
 class ChannelFollow:
     """Represents a channel follow."""
 
+    app: traits.RESTAware = attr.ib(repr=False, eq=False, hash=False, metadata={attr_extensions.SKIP_DEEP_COPY: True})
+    """The client application that models may use for procedures."""
+
     channel_id: snowflakes.Snowflake = attr.ib(eq=True, hash=True, repr=True)
     """The ID of the news channel that's being followed."""
 
     webhook_id: snowflakes.Snowflake = attr.ib(eq=True, hash=True, repr=True)
     """The ID of the webhook for this follow."""
 
+    async def fetch_channel(self) -> GuildChannel:
+        """Fetch the object of the guild channel being followed.
+
+        Returns
+        -------
+        hikari.channels.PartialChannel
+            The channel being followed. This will be a _derivative_ of
+            `hikari.channels.PartialChannel`, depending on the type of
+            channel you request for.
+
+            This means that you may get one of
+            `hikari.channels.PrivateTextChannel`,
+            `hikari.channels.GroupPrivateTextChannel`,
+            `hikari.channels.GuildTextChannel`,
+            `hikari.channels.GuildVoiceChannel`,
+            `hikari.channels.GuildStoreChannel`,
+            `hikari.channels.GuildNewsChannel`.
+
+            Likewise, the `hikari.channels.GuildChannel` can be used to
+            determine if a channel is guild-bound, and
+            `hikari.channels.TextChannel` can be used to determine
+            if the channel provides textual functionality to the application.
+
+            You can check for these using the `builtins.isinstance`
+            builtin function.
```

Comment:
I think you cant follow to DM's, group DM's, store nor voice guild channels

Reply:
that was just a copied doc, will change